src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWPApplication.cs

```diff
@@ -394,72 +378,277 @@ internal string LogoPathFromUri(string uri, string theme)
+                int appIconSize = 36;
+                var targetSizes = new List<int> { 16, 24, 30, 36, 44, 60, 72, 96, 128, 180, 256 }.AsParallel();
+                var pathFactorPairs = new Dictionary<string, int>();
 
-                        paths.Add(simplePath);
+                foreach (var factor in targetSizes)
+                {
+                    if (highContrast)
+                    {
+                        string suffixThemePath = $"{prefix}.targetsize-{factor}_{colorscheme}{extension}";
+                        string prefixThemePath = $"{prefix}.{colorscheme}_targetsize-{factor}{extension}";
                         paths.Add(suffixThemePath);
                         paths.Add(prefixThemePath);
-
-                        pathFactorPairs.Add(simplePath, factor);
                         pathFactorPairs.Add(suffixThemePath, factor);
                         pathFactorPairs.Add(prefixThemePath, factor);
                     }
-
-                    paths = paths.OrderByDescending(x => x.Contains(theme, StringComparison.OrdinalIgnoreCase)).ToList();
-                    var selectedIconPath = paths.OrderBy(x => Math.Abs(pathFactorPairs.GetValueOrDefault(x) - appIconSize)).FirstOrDefault(File.Exists);
-                    if (!string.IsNullOrEmpty(selectedIconPath))
-                    {
-                        return selectedIconPath;
-                    }
                     else
                     {
-                        ProgramLogger.LogException(
-                            $"|UWP|LogoPathFromUri|{Package.Location}" +
-                            $"|{UserModelId} can't find logo uri for {uri} in package location: {Package.Location}", new FileNotFoundException());
-                        return string.Empty;
+                        string simplePath = $"{prefix}.targetsize-{factor}{extension}";
+                        string altformUnPlatedPath = $"{prefix}.targetsize-{factor}_altform-unplated{extension}";
+                        paths.Add(simplePath);
+                        paths.Add(altformUnPlatedPath);
+                        pathFactorPairs.Add(simplePath, factor);
+                        pathFactorPairs.Add(altformUnPlatedPath, factor);
                     }
                 }
+
+                var selectedIconPath = paths.OrderBy(x => Math.Abs(pathFactorPairs.GetValueOrDefault(x) - appIconSize)).FirstOrDefault(File.Exists);
+                return selectedIconPath;
             }
             else
             {
-                ProgramLogger.LogException(
-                    $"|UWP|LogoPathFromUri|{Package.Location}" +
-                                                $"|Unable to find extension from {uri} for {UserModelId} " +
-                                                $"in package location {Package.Location}", new FileNotFoundException());
                 return string.Empty;
             }
         }
 
+        private bool SetColoredIcon(string path, string colorscheme)
```

Comment:
(Suggestion) I feel we can reuse a lot of code in this function and the SetHighContrastIcon function. 

One way that I can think of is to have a function which takes the logo type and logo path and sets those values so that we dont't have to do it every time. This can then be called from the getTargetIcon or getScaleIcon function as they have information about the logo type (high contrast or coloured). The getScaleIcon functions can simply return true or false about whether they are returning a valid logo or not.

Reply:
Thanks for the suggestion! I will modify the code.